docs/source/pytest.rst

```diff
@@ -0,0 +1,358 @@
+**************
+Pytest Support
+**************
+
+:py:mod:`cocotb_tools.pytest.plugin` provides full `pytest`_ integration with cocotb. Including:
+
+* `fixture`_ to cleanly setup and teardown cocotb tests and designs under test.
+* `plugins`_ that can extend cocotb testing capabilities.
+* `configuration`_ facilities to configure cocotb testing environment using command line arguments
+  ``--cocotb-*``, configuration files like `pyproject.toml`_ or `fixture`_ arguments for fine
+  control per test, class, module or session.
+* listing all available cocotb tests and their relationship with :py:mod:`cocotb_tools.runner`.
+* `marks`_ to easily set metadata on cocotb test functions.
+* filtering cocotb tests with `pytest`_ ``-k '<expression>'`` and ``-m '<markers>'`` options.
+* reporting all executed cocotb tests.
+* parallel execution of cococtb runners by using `pytest-xdist`_ plugin
+
+Enabling Plugin
+===============
+
+:py:mod:`cocotb_tools.pytest.plugin` can be enabled in various ways.
+
+When using `pyproject.toml`_ file (recommended way):
+
+.. code:: toml
+
+   [project.entry-points.pytest11]
```

Comment:
While the indentation for rST is 3 chars, code should still be indented with 4 chars (https://devguide.python.org/documentation/markup/#use-of-whitespace).